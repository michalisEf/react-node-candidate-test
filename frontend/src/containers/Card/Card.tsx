import React from 'react';

import './Card.css';

import { } from '../../components'
import { IJobCard } from '../../interfaces';

export const Card = (props: IJobCard): JSX.Element => {
	return (
		<div className="job-card">
			Start building your job card here
		</div>
	);
};
