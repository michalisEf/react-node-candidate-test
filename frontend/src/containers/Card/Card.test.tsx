import { render, screen } from '@testing-library/react';
import { Card } from './Card';

test('renders start text', () => {
  const props = {
    "id": "job-1000",
    "title": "React Developer",
    "location": "London",
    "salary": 75000,
    "type": "Permanent",
    "company": "Acme Company",
    "post_date": "14 days ago",
    "description": "React Developer Circa £75,000 City of London The React Developer will work on a re-platforming of existing products to take advantage … The React Developer will benefit from working in a purely greenfield project",
    "logo": "https://logo.clearbit.com/company.com"
  };
  render(<Card {...props} />);
  const app = screen.getByRole('heading');
  expect(app).toHaveTextContent('React Developer');
  expect(app).toMatchSnapshot();
});
