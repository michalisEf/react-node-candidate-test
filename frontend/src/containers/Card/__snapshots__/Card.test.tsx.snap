// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders start text 1`] = `
<h1>
  React Developer
</h1>
`;
